src/PoolManager.sol

```diff
@@ -203,34 +214,62 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         PoolId id = key.toId();
         _checkPoolInitialized(id);
 
-        key.hooks.beforeSwap(key, params, hookData);
-
-        uint256 feeForProtocol;
-        uint24 swapFee;
-        Pool.SwapState memory state;
-        (delta, feeForProtocol, swapFee, state) = pools[id].swap(
-            Pool.SwapParams({
-                tickSpacing: key.tickSpacing,
-                zeroForOne: params.zeroForOne,
-                amountSpecified: params.amountSpecified,
-                sqrtPriceLimitX96: params.sqrtPriceLimitX96
-            })
-        );
-
-        _accountPoolBalanceDelta(key, delta);
-
-        // the fee is on the input currency
-        unchecked {
-            if (feeForProtocol > 0) {
-                protocolFeesAccrued[params.zeroForOne ? key.currency0 : key.currency1] += feeForProtocol;
+        int128 hookDeltaInSpecified = key.hooks.beforeSwap(key, params, hookData);
```

Comment:
for lines 217 -227, shouldn't we be doing this only if the hook has the correct flag?

Reply:
Oh also could lines 1
217-224 be done in the hook library?? Might neaten up this outer level swap function